src/librustc_codegen_llvm/mir/operand.rs

```diff
@@ -112,32 +112,6 @@ impl OperandRef<'ll, 'tcx> {
                         );
                         OperandValue::Immediate(llval)
                     },
-                    layout::Abi::ScalarPair(ref a_scalar, ref b_scalar) => {
-                        let a_size = a_scalar.value.size(bx.tcx());
-                        let a = alloc.read_scalar(
-                            bx.tcx(), offset, a_size, a_scalar.value.align(bx.tcx()),
-                        ).map_err(econv)?;
-                        let b_align = b_scalar.value.align(bx.tcx());
-                        let b_offset = offset + a_size.abi_align(b_align);
-                        let b_size = b_scalar.value.size(bx.tcx());
-                        let b = alloc.read_scalar(
-                            bx.tcx(), b_offset, b_size, b_align,
-                        ).map_err(econv)?;
-                        let a_llval = scalar_to_llvm(
-                            bx.cx,
-                            a,
-                            a_scalar,
-                            layout.scalar_pair_element_llvm_type(bx.cx, 0, true),
-                        );
-                        let b_layout = layout.scalar_pair_element_llvm_type(bx.cx, 1, true);
-                        let b_llval = scalar_to_llvm(
-                            bx.cx,
-                            b,
-                            b_scalar,
-                            b_layout,
-                        );
-                        OperandValue::Pair(a_llval, b_llval)
-                    },
```

Comment:
I cannot make *any* connection between the title of this commit ("Enums can produce scalar pairs with undefs") and it just removing an entire match arm without replacement... but I am happy to see that code gone so I guess I won't question it. ;)

Reply:
It should never have been added. It is the same issue as why we have this normalization logic in https://github.com/rust-lang/rust/blob/master/src/librustc_mir/const_eval.rs#L105

An enum can be `ScalarPair`, but in the `None` case one of the variants can be `Undef`, which would fail here. `codegen` only needs `Scalar` for intrinsics, never `ScalarPair`, so I opted to just remove it entirely

Reply:
That's good, anything that kills this code is good. ;)